Wikipedia/Code/WMFArticleViewController.m

```diff
@@ -1651,6 +1651,22 @@ - (void)webViewController:(WebViewController *)controller didTapShareWithSelecte
     [self shareAFactWithTextSnippet:text];
 }
 
+- (void)webViewController:(WebViewController *)controller didTapEditMenuItemInMenuController:(UIMenuController *)menuController {
+    @weakify(self);
+    [self.webViewController.webView wmf_getSelectedTextEditInfoWithCompletionHandler:^(SelectedTextEditInfo* selectedTextEditInfo, NSError *error) {
+        @strongify(self);
+        if (error) {
+            return;
+        }
+        if (selectedTextEditInfo.isSelectedTextInTitleDescription) {
+            [self showTitleDescriptionEditor];
+        } else {
+            MWKSection *section = self.article.sections[selectedTextEditInfo.sectionID];
```

Comment:
might want to confirm the sections count before accessing

Reply:
hopefully there's no way to edit text if there's no section having that text... but can't hurt to check :)

fixed!